pkg/amqplib/AmqpConsumer.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace Enqueue\Amqplib;
+
+use Interop\Queue\InvalidMessageException;
+use Interop\Queue\PsrConsumer;
+use Interop\Queue\PsrMessage;
+use PhpAmqpLib\Channel\AMQPChannel;
+use PhpAmqpLib\Message\AMQPMessage as LibAMQPMessage;
+use PhpAmqpLib\Wire\AMQPTable;
+
+class AmqpConsumer implements PsrConsumer
+{
+    private $channel;
+    private $queue;
+
+    public function __construct(AMQPChannel $channel, AmqpQueue $queue)
+    {
+        $this->channel = $channel;
+        $this->queue = $queue;
+    }
+
+    public function getQueue()
+    {
+        return $this->queue;
+    }
+
+    public function receive($timeout = 0)
+    {
+        $end = microtime(true) + ($timeout / 1000);
```

Comment:
it would be good to add a `basic_consume` support. But it could be done at a later stage.